File: basicprogramming2.py
def basicprogramming2(A,N,t):
    if t == 1:
        s = set(A)
        for num in s:
            if 7777 - num in s:
                return 'Yes'
        return 'No'
    elif t == 2:
        return 'Unique' if len(set(A)) == N else 'Contains duplicate'
    elif t == 3:
        dict1 = {}
        for num in A:
            if num not in dict1:
                dict1[num] = 1
            else:
                dict1[num] += 1
        for key,value in dict1.items():
            if value > N/2:
                return key
        return -1
    elif t == 4:
        return sorted(A)[N//2] if N%2 else f'{sorted(A)[N//2 - 1]} {sorted(A)[N//2]}'
    elif t == 5:
        lst = []
        for num in A:
            if num in range(100,1000):
                lst.append(num)
        lst2 = list(map(str,sorted(lst)))
        return ' '.join(lst2)
        

integers = list(map(int,input().split()))
p1,p2 = integers[0],integers[1]
array = list(map(int,input().split()))
print(basicprogramming2(array,p1,p2))

File: basicprogramming2.java
import java.io.*;
import java.util.*;

class basicprogramming2 {
    public static void solve(ArrayList<Integer> A, int N, int t, PrintWriter pw) {
        if (t == 1) {
            for (int x: A) {
                if (Collections.binarySearch(A, 7777 - x) >= 0) {
                    pw.println("YES\n");
                    return;
                }
            }
            pw.println("NO\n");
        } else if (t == 2) {
            for (int i = 1; i < N; i++) {
                if (A.get(i - 1) == A.get(i)) {
                    pw.println("Contains duplicate");
                    return;
                }
            }
            pw.println("Unique");
        } else if (t == 3) {
            for (int i = 0; i < N; i++) {
                int counter = 1;
                while (i + 1 < N && A.get(i + 1) == A.get(i)) {
                    i++;
                    counter++;
                }
                if ((counter << 1) > N) {
                    pw.println(A.get(i));
                    return;
                }
            }
            pw.println(-1);
        } else if (t == 4) {
            if (N % 2 == 1) pw.println(A.get(N >> 1));
            else pw.println(A.get((N >> 1) - 1) + " " + A.get(N >> 1));
        } else {
            Boolean first = true;
            for (int v: A) {
                if (100 <= v && v <= 999) {
                    if (!first) pw.print(" ");
                    first = false;
                    pw.print(v);
                }
            }
            pw.println();
        }
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(System.out);
        String[] tokens = br.readLine().split(" ");
        int N = Integer.parseInt(tokens[0]), t = Integer.parseInt(tokens[1]);
        ArrayList<Integer> A = new ArrayList<>();
        tokens = br.readLine().split(" ");
        for (String elem: tokens) A.add(Integer.parseInt(elem));
        Collections.sort(A);
        solve(A,N,t,pw);
        br.close();
        pw.close();
    }
}